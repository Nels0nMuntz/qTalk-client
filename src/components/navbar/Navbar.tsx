import Link from 'next/link';
import React from 'react';
import { getServerSession } from 'next-auth';
import { Icons } from '../Icons';
import { buttonVariants } from '../ui/button';
import UserAccountNavbar from './UserAccountNavbar';
import { authoOptions } from '@/lib/auth';
import Searchbar from './Searchbar';

export default async function Navbar() {
  const session = await getServerSession(authoOptions);

  return (
    <header className="fixed top-0 inset-x-0 h-auto bg-zinc-100 border-b border-zinc-300 z-10 py-2">
      <div className="container max-w-7xl h-full mx-auto flex items-center justify-between gap-2">
        <Link href="/" className="flex gap-2 items-center">
          <Icons.logo className="w-8 h-8 sm:w-6 sm:h-6" />
          <span className="hidden md:block text-zinc-700 text-sm font-medium">
            qTalk
          </span>
        </Link>
        {session?.user && <Searchbar />}
        {session?.user ? (
          <UserAccountNavbar user={session.user} />
        ) : (
          <Link href="/sign-in" className={buttonVariants()}>
            Sign In
          </Link>
        )}
      </div>
    </header>
  );
}
