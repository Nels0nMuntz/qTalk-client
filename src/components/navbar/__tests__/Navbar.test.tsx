import { render, screen } from "@testing-library/react";
import { getServerSession } from 'next-auth';
import Navbar from "../Navbar";

jest.mock('next-auth', () => ({
  getServerSession: jest.fn()
}))
jest.mock('@/lib/auth', () => ({
  authoOptions: {}
}))
jest.mock('next/navigation', () => ({
  usePathname: jest.fn(), 
  useRouter: jest.fn(),
}))
jest.mock('@tanstack/react-query', () => ({
  useQuery: jest.fn(() => ({
    data: {},
    refetch: jest.fn(),
    isFetched: false,
  })), 
}))
jest.mock('axios', () => ({}))

describe("Navbar component", () => {
  it('renders correctly when user is not logged in', async () => {
    const getServerSessionMock = getServerSession as jest.MockedFunction<typeof getServerSession>;
    getServerSessionMock.mockResolvedValueOnce({
      user: null,
    });

    const NavbarSync = await Navbar();
    const { container } = render(NavbarSync)

    const signInLink = await screen.findByRole('link', { name: 'Sign In' })
    const searchInput = screen.queryByRole('combobox');
    const userAccountButton = screen.queryByRole('button');

    expect(container).toMatchSnapshot();
    expect(signInLink).toBeInTheDocument();
    expect(searchInput).not.toBeInTheDocument();
    expect(userAccountButton).not.toBeInTheDocument();
  });
  it('renders correctly when user is logged in', async () => {
    const getServerSessionMock = getServerSession as jest.MockedFunction<typeof getServerSession>;
    getServerSessionMock.mockResolvedValueOnce({
      user: {
        name: 'jenna',
        email: 'jenna@jenna.com',
        image: '/path/to/image'
      },
    });

    const NavbarSync = await Navbar()

    const { container } = render(NavbarSync)

    const searchInput = await screen.findByRole('combobox');
    const userAccountButton = await screen.findByRole('button');
    const signInLink = screen.queryByRole('link', { name: 'Sign In' })

    expect(container).toMatchSnapshot();
    expect(searchInput).toBeInTheDocument();
    expect(userAccountButton).toBeInTheDocument();
    expect(signInLink).not.toBeInTheDocument();
  });
})