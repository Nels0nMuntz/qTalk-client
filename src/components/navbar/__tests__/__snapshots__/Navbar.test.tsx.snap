// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly when user is logged in 1`] = `
<div>
  <header
    class="fixed top-0 inset-x-0 h-fit bg-zinc-100 border-b border-zinc-300 z-10 py-2"
  >
    <div
      class="container max-w-7xl h-full mx-auto flex items-center justify-between gap-2"
    >
      <a
        class="flex gap-2 items-center"
        href="/"
      >
        <svg
          class="w-8 h-8 sm:w-6 sm:h-6"
          fill="#000000"
          preserveAspectRatio="xMidYMid meet"
          version="1.1"
          viewBox="0 0 36 36"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <title>
            talk-bubbles-solid
          </title>
          <path
            class="clr-i-solid clr-i-solid-path-1"
            d="M8,19V11H5a3,3,0,0,0-3,3V32a1,1,0,0,0,.56.89,1,1,0,0,0,1-.1L8.71,29H22.15A2.77,2.77,0,0,0,25,26.13V25H14A6,6,0,0,1,8,19Z"
          />
          <path
            class="clr-i-solid clr-i-solid-path-2"
            d="M31,4H14a3,3,0,0,0-3,3V19a3,3,0,0,0,3,3H27.55l4.78,3.71a1,1,0,0,0,1,.11,1,1,0,0,0,.57-.9V7A3,3,0,0,0,31,4Z"
          />
          <rect
            fill-opacity="0"
            height="36"
            width="36"
            x="0"
            y="0"
          />
        </svg>
        <span
          class="hidden md:block text-zinc-700 text-sm font-medium"
        >
          qTalk
        </span>
      </a>
      <div
        class="flex h-full w-full flex-col bg-popover text-popover-foreground relative rounded-lg border max-w-lg z-50 overflow-visible"
        cmdk-root=""
      >
        <label
          cmdk-label=""
          for=":r2:"
          id=":r1:"
          style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        />
        <div
          class="flex items-center border-b px-3"
          cmdk-input-wrapper=""
        >
          <svg
            class="mr-2 h-4 w-4 shrink-0 opacity-50"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <path
              d="m21 21-4.3-4.3"
            />
          </svg>
          <input
            aria-autocomplete="list"
            aria-controls=":r0:"
            aria-expanded="true"
            aria-labelledby=":r1:"
            autocomplete="off"
            autocorrect="off"
            class="flex h-11 w-full rounded-md bg-transparent py-3 text-sm placeholder:text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50 outline-none border-none focus:border-none focus:outline-none ring-0"
            cmdk-input=""
            id=":r2:"
            placeholder="Search communities..."
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        data-state="closed"
        id="radix-:r3:"
        type="button"
      >
        <span
          class="relative flex shrink-0 overflow-hidden rounded-full h-8 w-8"
        >
          <span
            class="flex h-full w-full items-center justify-center rounded-full bg-muted"
          >
            <svg
              class="w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="12"
                cy="7"
                r="4"
              />
            </svg>
            <span
              class="sr-only"
            >
              jenna
            </span>
          </span>
        </span>
        <span
          class="sr-only"
        >
          Open menu
        </span>
      </button>
    </div>
  </header>
</div>
`;

exports[`Navbar component renders correctly when user is not logged in 1`] = `
<div>
  <header
    class="fixed top-0 inset-x-0 h-fit bg-zinc-100 border-b border-zinc-300 z-10 py-2"
  >
    <div
      class="container max-w-7xl h-full mx-auto flex items-center justify-between gap-2"
    >
      <a
        class="flex gap-2 items-center"
        href="/"
      >
        <svg
          class="w-8 h-8 sm:w-6 sm:h-6"
          fill="#000000"
          preserveAspectRatio="xMidYMid meet"
          version="1.1"
          viewBox="0 0 36 36"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <title>
            talk-bubbles-solid
          </title>
          <path
            class="clr-i-solid clr-i-solid-path-1"
            d="M8,19V11H5a3,3,0,0,0-3,3V32a1,1,0,0,0,.56.89,1,1,0,0,0,1-.1L8.71,29H22.15A2.77,2.77,0,0,0,25,26.13V25H14A6,6,0,0,1,8,19Z"
          />
          <path
            class="clr-i-solid clr-i-solid-path-2"
            d="M31,4H14a3,3,0,0,0-3,3V19a3,3,0,0,0,3,3H27.55l4.78,3.71a1,1,0,0,0,1,.11,1,1,0,0,0,.57-.9V7A3,3,0,0,0,31,4Z"
          />
          <rect
            fill-opacity="0"
            height="36"
            width="36"
            x="0"
            y="0"
          />
        </svg>
        <span
          class="hidden md:block text-zinc-700 text-sm font-medium"
        >
          qTalk
        </span>
      </a>
      <a
        class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
        href="/sign-in"
      >
        Sign In
      </a>
    </div>
  </header>
</div>
`;
